ext/standard/array.c

```diff
@@ -3168,8 +3168,8 @@ PHP_FUNCTION(array_push)
 
 		if (zend_hash_next_index_insert(Z_ARRVAL_P(stack), &new_var) == NULL) {
 			Z_TRY_DELREF(new_var);
-			php_error_docref(NULL, E_WARNING, "Cannot add element to the array as the next element is already occupied");
-			RETURN_FALSE;
+			zend_throw_error(NULL, "Cannot add element to the array as the next element is already occupied");
```

Comment:
So, this is actually a pretty generic warning, in particular also used by `$foo[] = $x`, see https://github.com/php/php-src/blob/0038db22510cce37b24491c266123ba5b1e93953/Zend/zend_execute.c#L1971. I think whatever we do here, we should have a consistent behavior between `$foo[] = $x` and `array_push($foo, $x)`.